api/src/it/kotlin/eu/codeloop/thehub/sync/FetchFromAPITest.kt

```diff
@@ -0,0 +1,112 @@
+package eu.codeloop.thehub.sync
+
+import com.github.tomakehurst.wiremock.client.WireMock.*
+import eu.codeloop.thehub.IntegrationTest
+import eu.codeloop.thehub.base.DatabaseSetup
+import eu.codeloop.thehub.base.DatabaseSetupOperations
+import org.awaitility.Awaitility.await
+import org.awaitility.core.ConditionFactory
+import org.junit.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.MediaType.*
+import org.springframework.test.web.servlet.ResultActions
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import java.util.concurrent.TimeUnit.SECONDS
+import wiremock.org.apache.http.HttpHeaders.CONTENT_TYPE
+
+class FetchFromAPITest : IntegrationTest() {
+
+    @Autowired
+    private lateinit var databaseSetup: DatabaseSetup
+
+    @Value("\${thehub.sync.domains}")
+    private lateinit var syncDomains: String
+
+    @Test
+    fun `it should fetch jobs from external api and return sorted list`() {
+        // given
+        databaseSetup.prepareDatabase(
+            DatabaseSetupOperations.deleteAll()
+        )
+        removeAllMappings()
+        addJsonMapping("/api/jobs?page=1", "body-api-jobs-yH61E.json")
+        addHtmlMapping("/jobs/chief-financial-officer-cfo-1", "body-jobs-chief-financial-officer-cfo-1-kgwWj.txt")
+        addHtmlMapping("/jobs/chief-marketing-officer-cmo-6", "body-jobs-chief-marketing-officer-cmo-6-8G7JB.txt")
+        val syncRequest = MockMvcRequestBuilders
+            .post("/api/jobs/sync")
+            .accept(ALL_VALUE)
+        val jobsRequest = MockMvcRequestBuilders
+            .get("/api/jobs")
+            .accept(APPLICATION_JSON_VALUE)
+
+        // when
+        val syncResult = mvc.perform(syncRequest)
+        val jobsResult = await().atMost(10, SECONDS).untilRequest({ mvc.perform(jobsRequest) }) {
+            response -> response.andExpect(jsonPath("$.totalElements").value(2))
+        }
+
+        // then
+        val chiefMarketingIndex = 0
+        val chiefFinancialIndex = 1
+        syncResult
+            .andExpect(status().isNoContent)
+        jobsResult
+            .andExpect(status().isOk)
+            .andExpect(jsonPath("$.totalPages").value(1))
+            .andExpect(jsonPath("$.number").value(1))
+            .andExpect(jsonPath("$.size").isNumber)
+            .andExpect(jsonPath("$.first").value(true))
+            .andExpect(jsonPath("$.last").value(true))
+            .andExpect(jsonPath("$.totalElements").value(2))
+            .andExpect(jsonPath("$.numberOfElements").value(2))
+            .andExpect(jsonPath("$.data[${chiefMarketingIndex}].jobId").value("chief-marketing-officer-cmo-6"))
```

Comment:
replace `chiefMarketingIndex` with `0` and `chiefFinancialIndex` with `1`. `.andExpect(jsonPath("$.data[${chiefFinancialIndex}].jobId")` does not look natural